iothub/device/src/DirectMethod/DirectMethodResponse.cs

```diff
@@ -39,7 +41,89 @@ public DirectMethodResponse(int status)
         /// </summary>
         protected internal PayloadConvention PayloadConvention { get; set; }
 
-        internal byte[] GetPayloadObjectBytes()
+        /// <summary>
+        /// The direct method response payload, deserialized to the specified type.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Use this method when the payload type is known and it can be deserialized using the configured
+        /// <see cref="PayloadConvention"/>. If it is not JSON or the type is not known, use <see cref="GetPayloadAsBytes"/>.
+        /// </para>
+        /// <para>
+        /// One intended usage of this method is to deserialize the direct method response received by an edge module client
+        /// after it invokes a direct method on an edge device or an edge module connected to the same edge hub.
+        /// These operations are invoked using the API <see cref="IotHubModuleClient.InvokeMethodAsync(string, EdgeModuleDirectMethodRequest, System.Threading.CancellationToken)"/>
+        /// and <see cref="IotHubModuleClient.InvokeMethodAsync(string, string, EdgeModuleDirectMethodRequest, System.Threading.CancellationToken)"/>.
+        /// </para>
+        /// </remarks>
+        /// <typeparam name="T">The type to deserialize the direct method response payload to.</typeparam>
+        /// <param name="payload">When this method returns true, this contains the value of the direct method response payload.
+        /// When this method returns false, this contains the default value of the type <c>T</c> passed in.</param>
+        /// <returns><c>true</c> if the direct method response payload can be deserialized to type <c>T</c>; otherwise, <c>false</c>.</returns>
+        /// <example>
+        /// <code language="csharp">
+        /// DirectMethodResponse response = await client
+        ///     .InvokeMethodAsync(deviceId, moduleId, directMethodRequest, cancellationToken)
+        ///     .ConfigureAwait(false);
+        /// if (response.TryGetPayload(out MyCustomType customTypePayload))
+        /// {
+        ///     // do work
+        ///     // ...
+        /// }
+        /// else
+        /// {
+        ///     // Get the payload as bytes
+        ///     byte[] arr = directMethodRequest.GetPayloadAsBytes();
+        ///
+        ///     // deserialize as needed and do work...
+        /// }
+        ///
+        /// // ...
+        /// </code>
+        /// </example>
+        public bool TryGetPayload<T>(out T payload)
+        {
+            payload = default;
+
+            try
+            {
+                payload = PayloadConvention.GetObject<T>(GetPayloadAsBytes());
```

Comment:
Why not just:

```suggestion
                payload = PayloadConvention.GetObject<T>(Payload);
```

Reply:
GetObject<T>() requires the input to be bytes. Payload is of type `object`.

`public abstract T GetObject<T>(byte[] objectToConvert);`